packages/slides-frontend/src/containers/redux-store/DeckReducer/reducer.ts

```diff
@@ -47,6 +84,37 @@ const DeckState = (state: Deck = initialDeck, action: Action): Deck =>
         } else alert('Not possible to remove the only slide');
         break;
       }
+      case CLONE_SLIDE: {
+        // create a new blank slide
+        let slide:Slide = {
+          ID: uuidv4(),
+          itemsArray: [],
+        }
+        // with a loop over the current itemsArray, create and push newItems in the new itemsArray
+        slide.itemsArray = draft.slides[draft.currentSlide].itemsArray.map(currentItem => {
+          const copyObj = JSON.parse(JSON.stringify(currentItem));
+          if(copyObj.type === ItemTypes.TEXT) {
+            const itemObj: Text = new Text();
```

Comment:
It would be nice if `Text` had a `clone` method.

Reply:
In the definitions I have a `newItem()` method, that gets an object and tries to create the appropriate `Text` or `Image` type.
I could use this instead, but I had some issues because I use TypeScript.
it would be nice to do:
`const itemObj: Text = newItem({type: "TEXT"})`
but TypeScript complains, because the newItem returns the mother class `Item`.
I need to have it as Text object in the reducer at the CLONE_SLIDE case in order to access its `Data` attribute.